### PR TITLE
Update Flutter pipelines

### PR DESCRIPTION
**Pull request by nicolas-chaix-didomi**
Closes: CMP-2101

- update native sdks (pending new releases)
- update android build setup
- implement new events `showPreferences` / `hidePreferences`
- split UI test pipeline to be able to run android and ios UI tests independently
- update / fix ios pipeline for new Mac mini M1
- update project's readme file

### CODE REVIEW

File: .github/scripts/uitests_for_ios.sh [hunk 1]
```diff
@@ -54,13 +55,19 @@ for file in $(find integration_test -maxdepth 1 -type f); do
   zip -r "ios_tests.zip" "Release-iphoneos" "Runner_iphoneos$dev_target-arm64-armv7.xctestrun"
   popd
 
+#  echo "--------------------------------------------------------"
```
Comment (pmerlet-at-didomi) [inline .github/scripts/uitests_for_ios.sh:58]: Should we keep this commented code?

File: example/android/gradlew [hunk 2]
```diff
@@ -134,27 +156,30 @@ if $cygwin ; then
...
-        (1) set -- "$args0" ;;
-        (2) set -- "$args0" "$args1" ;;
-        (3) set -- "$args0" "$args1" "$args2" ;;
-        (4) set -- "$args0" "$args1" "$args2" "$args3" ;;
-        (5) set -- "$args0" "$args1" "$args2" "$args3" "$args4" ;;
-        (6) set -- "$args0" "$args1" "$args2" "$args3" "$args4" "$args5" ;;
-        (7) set -- "$args0" "$args1" "$args2" "$args3" "$args4" "$args5" "$args6" ;;
-        (8) set -- "$args0" "$args1" "$args2" "$args3" "$args4" "$args5" "$args6" "$args7" ;;
-        (9) set -- "$args0" "$args1" "$args2" "$args3" "$args4" "$args5" "$args6" "$args7" "$args8" ;;
+        0) set -- ;;
```
Comment (pmerlet-at-didomi) [inline example/android/gradlew:162]: Is this change on purpose? The syntax looks strange

File: .github/scripts/uitests_for_ios.sh (hunk 1 above)
Comment (nicolas-chaix-didomi) [reply .github/scripts/uitests_for_ios.sh:58]: yes, I kept it for local use of the script

File: example/android/gradlew (hunk 2 above)
Comment (nicolas-chaix-didomi) [reply example/android/gradlew:162]: this comes from the gradle wrapper update...

**Review by pmerlet-at-didomi — APPROVED** [review]

**Review by felipe-saez — APPROVED** [review]
